cognite_toolkit/_cdf_tk/client/data_classes/charts_data.py

```diff
@@ -0,0 +1,303 @@
+from cognite.client.data_classes.data_modeling import NodeId, ViewId
+from cognite.client.utils._auxiliary import to_camel_case
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+
+@dataclass
+class ChartObject(CogniteObject):
+    # ChartObjects are used in the frontend and the backend does not do any validation of these fields.
+    # Therefore, to ensure that we do not lose any data, we store unknown fields in a separate dictionary.
+    # This allows unknown fields to be preserved when loading and dumping ChartObjects
+    # (serialization and deserialization).
+    _unknown_fields: dict[str, object] | None = field(default=None, init=False, repr=False)
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        """Load a ChartObject from a dictionary."""
+        instance = super()._load(resource, cognite_client=cognite_client)
+        known_camel_case_props = cls._known_camel_case_props()
+        instance._unknown_fields = {k: v for k, v in resource.items() if k not in known_camel_case_props}
+        return instance
+
+    @classmethod
+    @lru_cache(maxsize=1)
+    def _known_camel_case_props(cls) -> set[str]:
+        return {to_camel_case(f.name) for f in fields(cls)}
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        """Dump the ChartObject to a dictionary."""
+        data = super().dump(camel_case=camel_case)
+        if self._unknown_fields:
+            data.update(self._unknown_fields)
+        return data
+
+
+@dataclass
+class UserInfo(ChartObject):
+    id: str | None = None
+    email: str | None = None
+    display_name: str | None = None
+
+
+@dataclass
+class ChartSettings(ChartObject):
+    show_y_axis: bool = True
+    show_min_max: bool = True
+    show_gridlines: bool = True
+    merge_units: bool = False
+
+
+@dataclass
+class ThresholdFilter(ChartObject):
+    min_unit: str | None = None
+    max_unit: str | None = None
+
+
+@dataclass
+class ChartCall(ChartObject):
+    id: str | None = None
+    hash: int | None = None
+    call_id: str | None = None
+    call_date: int | None = None
+    status: str | None = None
+
+
+@dataclass
+class SubSetting(ChartObject):
+    auto_align: bool | None = None
+
+
+@dataclass
+class FlowElement(ChartObject):
+    id: str | None = None
+    type: str | None = None
+    position: tuple[float | None, float | None] | None = None
+    data: dict[str, object] | None = None
+
+
+@dataclass
+class Flow(ChartObject):
+    zoom: float | None = None
+    elements: list[FlowElement] | None = None
+    position: tuple[float | None, float | None] | None = None
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        data = super().dump(camel_case=camel_case)
+        if self.elements:
+            data["elements"] = [el.dump(camel_case=camel_case) for el in self.elements]
```

Comment:
Doesn't super.dump propogate the camel_case parameter down to FlowElement?

Reply:
No, unfortunately not Pydantic would be stronger.